cpp/include/raft/neighbors/nn_descent_types.hpp

```diff
@@ -133,6 +155,16 @@ struct index : ann::index {
     return graph_view_;
   }
 
+  /** neighborhood graph distances [size, graph-degree] */
+  [[nodiscard]] inline auto distances() noexcept -> device_matrix_view<T, int64_t, row_major>
+  {
+    if (distances_view_.has_value()) {
+      return distances_view_.value();
+    } else {
+      return raft::make_device_matrix<T, int64_t>(res_, 0, 0).view();
+    }
+  }
```

Comment:
Let's just return the optional directly here
```suggestion
  [[nodiscard]] inline auto distances() noexcept -> std::optional<device_matrix_view<T, int64_t, row_major>>
  {
    return distances_view_;
  }
```

Reply:
Wouldn't this require callers to this `distances()` function to take care of the optional?? (Like calling `index.distances().value()` instead of just calling `index.distances()`)?

Reply:
Yes, it would. But IMO it is the correct approach because the distances are actually optional in this index, and returning an optional conveys that.